routines.go

```diff
@@ -277,3 +280,165 @@ func OrderbookUpdaterRoutine() {
 		time.Sleep(time.Second * 10)
 	}
 }
+
+// WebsocketRoutine Initial routine management system for websocket
+func WebsocketRoutine() {
+	log.Println("Connecting exchange Websocket services...")
+
+	for i := range bot.exchanges {
+		go func(i int) {
+			log.Printf("Establishing websocket connection for %s",
+				bot.exchanges[i].GetName())
+
+			ws, err := bot.exchanges[i].GetWebsocket()
+			if err != nil {
+				return
+			}
+
+			// Data handler routine
+			go WebsocketDataHandler(ws)
+
+			err = ws.Connect()
+			if err != nil {
+				switch err.Error() {
+				case exchange.WebsocketNotEnabled:
+					// Store in memory if enabled in future
+				default:
+					log.Println(err)
+				}
+			}
+		}(i)
+	}
+}
+
+var shutdowner = make(chan struct{}, 1)
+var wg sync.WaitGroup
+
+// Websocketshutdown shuts down the exchange routines and then shuts down
+// governing routines
+func Websocketshutdown(ws *exchange.Websocket) error {
+	err := ws.Shutdown() // shutdown routines on the exchange
+	if err != nil {
+		log.Fatalf("routines.go error - failed to shutodwn %s", err)
+	}
+
+	timer := time.NewTimer(5 * time.Second)
+	c := make(chan struct{}, 1)
+
+	go func(c chan struct{}) {
+		close(shutdowner)
+		wg.Wait()
+		c <- struct{}{}
+	}(c)
+
+	select {
+	case <-timer.C:
+		return errors.New("routines.go error - failed to shutdown routines")
+
+	case <-c:
+		return nil
+	}
+}
+
+// streamDiversion is a diversion switch from websocket to REST or other
+// alternative feed
+func streamDiversion(ws *exchange.Websocket) {
+	wg.Add(1)
+	defer wg.Done()
+
+	for {
+		select {
+		case <-shutdowner:
+			return
+
+		case <-ws.Connected:
+			log.Println("EXCHANGE WEBSOCKET ACTIVATED________________________________")
```

Comment:
Need a config var or command line flag for websocket verbosity otherwise too much info will be logged. By default it should be limited.